src/databricks/labs/ucx/source_code/base.py

```diff
@@ -178,12 +178,13 @@ class CurrentSessionState:
 
     @classmethod
     def from_json(cls, json: dict) -> CurrentSessionState:
+        mode_str = json.get('data_security_mode', None)
         return cls(
             schema=json.get('schema', DEFAULT_SCHEMA),
             catalog=json.get('catalog', DEFAULT_CATALOG),
             spark_conf=json.get('spark_conf', None),
             named_parameters=json.get('named_parameters', None),
-            data_security_mode=json.get('data_security_mode', None),
+            data_security_mode=compute.DataSecurityMode(mode_str) if mode_str else None,
```

Comment:
this may crash when we introduce a new data security mode and old clients are not upgraded to the newer SDK. make it a bit more robust.